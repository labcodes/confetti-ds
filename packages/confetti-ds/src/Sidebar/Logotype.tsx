import React from "react";
import PropTypes from "prop-types";

/**
 *
 * @param logoSrc Source of the logo to be rendered.
 * @param altText Text that will specify the HTML alt attribute of an <img> element.
 * @returns {JSX.Element}
 * @constructor
 */
export default function Logotype({ logoSrc, altText }) {
  return (
    <div className="lab-narrow-sidebar__logo">
      <img src={logoSrc} alt={altText} />
    </div>
  );
}

Logotype.propTypes = {
  /** Source of the logo to be rendered. */
  logoSrc: PropTypes.string.isRequired,
  /** Text that will specify the HTML alt attribute of an <img> element. */
  altText: PropTypes.string.isRequired,
};
