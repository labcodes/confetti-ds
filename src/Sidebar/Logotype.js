import React from "react";
import PropTypes from "prop-types";

export default class Logotype extends React.Component {
  static propTypes = {
    /** Source of the logo to be rendered. */
    logoSrc: PropTypes.string.isRequired,
    /** Text that will specify the HTML alt attribute of an <img> element. */
    altText: PropTypes.string.isRequired,
  };

  render() {
    const { logoSrc, altText } = this.props;
    return (
      <div className="lab-narrow-sidebar__logo">
        <img src={logoSrc} alt={altText} />
      </div>
    );
  }
}
