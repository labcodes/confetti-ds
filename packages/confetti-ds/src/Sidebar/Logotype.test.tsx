import React from "react";
import renderer from "react-test-renderer";

import Logotype from "./Logotype";

describe("Logotype", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<Logotype logoSrc="src" altText="text" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
