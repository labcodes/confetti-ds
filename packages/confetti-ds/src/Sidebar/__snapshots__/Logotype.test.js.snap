// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logotype renders with base props 1`] = `
<div
  className="lab-narrow-sidebar__logo"
>
  <img
    alt="text"
    src="src"
  />
</div>
`;
